woudc_data_registry/models.py

```diff
@@ -904,6 +904,94 @@ def __repr__(self):
         return 'DataRecord({}, {})'.format(self.data_record_id, self.url)
 
 
+class Contribution():
+    """
+    Data Registry Contribution
+
+    """
+    __tablename__ = 'contributions'
+
+    id_field = 'contribution_id'
+    id_dependencies = ['project_id', 'dataset_id', 'station_id',
+                       'instrument_name']
+
+    project_id = Column(String, ForeignKey('projects.project_id'),
+                        nullable=False)
+    contribution_id = Column(String, primary_key=True)
+    dataset_id = Column(String, ForeignKey('datasets.dataset_id'),
+                        nullable=False)
+    station_id = Column(String, ForeignKey('stations.station_id'),
+                        nullable=False)
+    station_name = Column(String, nullable=False)
+    country_id = Column(String, nullable=False)
+    country_name_en = Column(String, nullable=False)
+    country_name_fr = Column(String, nullable=False)
+    instrument_id = Column(String, nullable=False)
+    instrument_name = Column(String,
+                             ForeignKey('instruments.name'), nullable=False)
+    instrument_model = Column(String, nullable=False)
+    start_date = Column(Date, nullable=False)
+    end_date = Column(Date, nullable=False)
+    x = Column(Float, nullable=False)
+    y = Column(Float, nullable=False)
+    z = Column(Float, nullable=False)
+
+    def __init__(self):
+
+        self.project_id = dict()
+        self.contribution_id = ""
+        self.dataset_id = dict()
+        self.station_id = dict()
+        self.country_id = dict()
+        self.country_name_en = dict()
+        self.country_name_fr = dict()
+        self.instrument_id = dict()
+        self.instrument_name = dict()
+        self.instrument_model = dict()
```

Comment:
Not sure that instrument model should be part of this class, you can always take it out now and add it back in later.

I say that because not having model would make it more convenient to search for things.
Besides, as long as there's no `instrument_model` in the identifier field, you could have multiple instrument models that give the same identifier which kind of goes against what the identifier is supposed to do.